beesdoo_base/models/partner.py

```diff
@@ -49,19 +51,15 @@ def write(self, values):
                 if command[0] == 2:
                     command[0] = 3
         return super(Partner, self).write(values)
+    
+    @api.multi
+    def _deactivate_active_cards(self):
+        if len(self.member_card_ids) > 0:
+            for c in self.member_card_ids:
+                if c.valid:
+                    c.valid = False
+                    c.end_date = fields.Date.today()      
+    @api.multi            
+    def _new_card(self, txt):
+        self.env['member.card'].create({'partner_id' : self.env.context['active_id'],'comment' : txt})
```

Comment:
Need a parameter to get the user_id since this method is called with sudo and we loose the information of the user
    @api.multi
    def _new_card(self, reason, user_id):
        self.env['member.card'].create({'partner_id' : self.id,'responsible_id' : user_id, 'comment' : reason})